server/controllers/files.py

```diff
@@ -40,4 +40,10 @@ def file_url(file_id):
                                                    .format(ext_file.filename))
         return response
     else:
-        return redirect(storage.get_blob_url(storage_obj.name))
+        postpend = ''
+        if request.args.get('raw'):
+            postpend = '&response-content-type={}'.format(ext_file.mimetype)
+        elif request.args.get('download'):
+            postpend = '&response-content-disposition=attachment; filename={0!s}'.format('ext_file.filename')
```

Comment:
This filename should be `os.path.basename(ext_file.filename)` instead of the string `'ext_file.filename'`. Also, use `urllib.urlparse.urlencode` to encode the query string in case the filename has characters that must be escaped.